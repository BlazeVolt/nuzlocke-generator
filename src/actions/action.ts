export interface Action<T = 'NO_ACTION_PROVIDED'> {
    type: T;
    [payload: string]: any;
    readonly meta?: any;
}
