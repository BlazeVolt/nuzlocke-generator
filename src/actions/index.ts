export { Action } from './action';
export { saveNuzlocke, SAVE_NUZLOCKE } from './saveNuzlocke';
export { addPokemon, ADD_POKEMON } from './addPokemon';
export { deletePokemon, DELETE_POKEMON } from './deletePokemon';
export { editPokemon, EDIT_POKEMON } from './editPokemon';
export { editBox, EDIT_BOX } from './editBox';
export { selectPokemon, SELECT_POKEMON } from './selectPokemon';
export { modifyDeletionConfirmation, MODIFY_DELETION_CONFIRMATION } from './modifyDeletionConfirmation';
export { editGame, EDIT_GAME } from './editGame';
export { editTrainer, EDIT_TRAINER } from './editTrainer';