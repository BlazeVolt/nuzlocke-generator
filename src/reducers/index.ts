import { applyMiddleware, combineReducers } from 'redux';
import createHistory from 'history/createBrowserHistory';
import { routerReducer } from 'react-router-redux';

import { nuzlocke } from './nuzlocke';
import { pokemon } from './pokemon';
import { box } from './box';
import { confirmation } from './confirmation';
import { selectedId } from './selectedId';
import { game } from './game';
import { trainer } from './trainer';
import { history } from './history';
import { style } from './style';

// export const appReducers = combineReducers({
//   nuzlocke,
//   pokemon
// });

export const appReducers = combineReducers({
    nuzlocke,
    pokemon,
    box,
    selectedId,
    confirmation,
    game,
    trainer,
    history,
    style,
    router: routerReducer,
});

export const reducers = {
    box,
    confirmation,
    game,
    nuzlocke,
    pokemon,
    selectedId,
    trainer,
    history,
    style,
    router: routerReducer,
};
