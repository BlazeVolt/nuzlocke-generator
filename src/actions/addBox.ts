import { Action } from './action';

export type ADD_BOX = 'ADD_BOX';
export const ADD_BOX: ADD_BOX = 'ADD_BOX';

export interface AddBoxArgs {
    name: string;
    background: string;
    inheritFrom: 'Team' | 'Boxed' | 'Dead' | 'Champs';
}

export type addBox = ({ name, background, inheritFrom }: AddBoxArgs) => Action<ADD_BOX>;
export function addBox({ name, background, inheritFrom }: AddBoxArgs): Action<ADD_BOX> {
    return {
        type: ADD_BOX,
        name,
        background,
        inheritFrom,
    };
}
