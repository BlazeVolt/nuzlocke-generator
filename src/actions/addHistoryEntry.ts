import { Action } from './action';
import { HistoryEntry } from 'models';

export type ADD_HISTORY_ENTRY = 'ADD_HISTORY_ENTRY';
export const ADD_HISTORY_ENTRY: ADD_HISTORY_ENTRY = 'ADD_HISTORY_ENTRY';

export type addHistoryEntry = (he: HistoryEntry) => Action<ADD_HISTORY_ENTRY>;
export const addHistoryEntry = (historyEntry: HistoryEntry): Action<ADD_HISTORY_ENTRY> => {
    return {
        type: ADD_HISTORY_ENTRY,
        historyEntry,
    };
};
