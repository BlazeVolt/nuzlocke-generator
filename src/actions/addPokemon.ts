import { Action } from './action';
import { Pokemon } from '../models';

export type ADD_POKEMON = 'ADD_POKEMON';
export const ADD_POKEMON: ADD_POKEMON = 'ADD_POKEMON';

export const addPokemon = (pokemon: Pokemon): Action<ADD_POKEMON> => {
    return {
        type: ADD_POKEMON,
        pokemon,
    };
};
