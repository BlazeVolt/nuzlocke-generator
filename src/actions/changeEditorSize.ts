import { Action } from './action';

export type CHANGE_EDITOR_SIZE = 'CHANGE_EDITOR_SIZE';
export const CHANGE_EDITOR_SIZE: CHANGE_EDITOR_SIZE = 'CHANGE_EDITOR_SIZE';

export type changeEditorSize = (mode: boolean) => Action<CHANGE_EDITOR_SIZE>;
export const changeEditorSize = (mode: boolean): Action<CHANGE_EDITOR_SIZE> => {
    return {
        type: CHANGE_EDITOR_SIZE,
        mode,
    };
};
