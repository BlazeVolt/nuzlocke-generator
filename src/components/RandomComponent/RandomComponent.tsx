import * as React from 'react';
import { connect } from 'react-redux';
import { css, cx } from 'emotion';
import { Action } from 'actions';

import {
    modifyDeletionConfirmation
} from 'actions';
import { reducers } from 'reducers';

export const RandomComponentStyles = {
    counter: css`
        background: white;
        color: red;
        display: flex;
        padding: .25rem;
    `,
    counterNumber: css`
        font-size: 1.25rem;
    `,
    footer: css`
        background: lightblue;
        color: #222;
        padding: .5rem;
    `,
    modifier: css`
        background: white;
        color: red;
    `
};


export interface RandomComponentProps {
    onClick(e?: React.MouseEvent<HTMLElement>): void;
    modifyDeletionConfirmation: modifyDeletionConfirmation;
    confirmation: typeof reducers.confirmation;
}

export interface RandomComponentState {
    count: number;
}

export class RandomComponentBase extends React.Component<RandomComponentProps, RandomComponentState> {
    public state = {
        count: 0,
    };

    public componentWillMount() {
        console.log(`Ran componentWillMount`);
    }

    public componentDidMount() {
        console.log(`Ran componentDidMount`);
        this.props.modifyDeletionConfirmation(false);
    }

    private onMinusClick = e => this.setState({ count: this.state.count - 1});

    private onPlusClick = e => this.setState({ count: this.state.count + 1});

    public render() {
        const { modifyDeletionConfirmation, onClick } = this.props;
        return (
            <div className={cx(RandomComponentStyles.counter)}>
                <div id='b1' role='button' onClick={this.onMinusClick}>-1</div>
                <div className={cx('clicks', RandomComponentStyles.counterNumber)}>
                    { this.state.count }
                </div>
                <div id='b2' role='button' onClick={this.onPlusClick}>+1</div>
                <div className={cx(RandomComponentStyles.footer)}>
                    <button id='click-this' onClick={onClick}>Click This</button>
                    <div id='modifier' role='button'>Modify</div>
                    { this.props.children ? this.props.children : <div>No Children Provided</div> }
                </div>
            </div>
        );
    }
}

export const RandomComponent = connect(
    (state: Partial<typeof reducers>) => ({
        confirmation: state.confirmation,
    }),
    {
        modifyDeletionConfirmation
    }
)(RandomComponentBase);