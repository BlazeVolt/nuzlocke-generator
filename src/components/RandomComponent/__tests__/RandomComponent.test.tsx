import * as React from 'react';
import { shallow, mount } from 'enzyme';
import * as emotion from 'emotion';
import configureStore from 'redux-mock-store';

import { RandomComponentBase, RandomComponent, RandomComponentProps } from '../RandomComponent';
import { Action, modifyDeletionConfirmation, MODIFY_DELETION_CONFIRMATION } from 'actions';
import { reducers } from 'reducers';


const props: RandomComponentProps = {
    onClick: e => {},
    modifyDeletionConfirmation: b => ({ type: MODIFY_DELETION_CONFIRMATION, newStatus: b }),
    confirmation: reducers.confirmation,
};

describe(`<${RandomComponentBase.name} />` , () => {
    it('shallow versus mount', () => {

        const child = <div>
            <div>
                <div>{'Child'}</div>
                <RandomComponentBase {...props}></RandomComponentBase>
            </div>
        </div>;

        const shallowWrapper = shallow(<RandomComponentBase {...props} >{ child }</RandomComponentBase>);
        const mountWrapper = mount(<RandomComponentBase {...props}>{ child }</RandomComponentBase>);

        console.log(shallowWrapper.debug());
        expect(shallowWrapper.text()).toContain('Child');
        expect(shallowWrapper.text()).not.toContain('No Children Provided');


        console.log(mountWrapper.debug());
        expect(mountWrapper.text()).toContain('Child');
        expect(mountWrapper.text()).toContain('No Children Provided');

    });

    it('tests children', () => {
        const wrapperWithChildren = mount(
            <RandomComponentBase {...props}>
                <div className='child-1'>
                    <div className='child-1-1'>Child 1</div>
                    <div className='child-1-2'>Child 2</div>
                </div>
                <div className='child-2'></div>
            </RandomComponentBase>
        );

        const wrapperWithoutChildren = mount(<RandomComponentBase {...props} />);

        expect(wrapperWithChildren.childAt(0).text()).toContain('Child 1');
        expect(wrapperWithoutChildren.text()).toContain('No Children Provided');
    });

    it('renders the clicks section', () => {
        const wrapper = mount(<RandomComponentBase {...props} />);
        expect(wrapper.find('.clicks')).toMatchSnapshot();
    });

});

describe(`<${RandomComponent.displayName} />`, () => {
    const middlewares = [];
    const mockStore = configureStore(middlewares);

    beforeEach(() => {
        const initialState = {};
        this.store = mockStore(initialState);
    });

    it('should dispatch actions', () => {

        this.store.dispatch(modifyDeletionConfirmation(true));

        const actions = this.store.getActions();
        const expected = { type: MODIFY_DELETION_CONFIRMATION, newStatus: true };

        expect(actions).toEqual([expected]);
    });

    it('simulates clicks', () => {
        const RC: React.ComponentClass<RandomComponentProps & { store: any }> = RandomComponent as any;
        const click = jest.fn();

        const wrapper = mount(<RC onClick={click} confirmation={reducers.confirmation} store={this.store} modifyDeletionConfirmation={modifyDeletionConfirmation} />);

        wrapper.find('#click-this').simulate('click');

        expect(click.mock.calls.length).toBe(1);
    });

    it('simulates clicks + counter', () => {
        const RC: React.ComponentClass<RandomComponentProps & { store: any }> = RandomComponent as any;

        const wrapper = mount(<RC {...props} store={this.store} />);

        expect(wrapper.find('.clicks').text()).toBe('0');

        wrapper.find('#b1').simulate('click');

        expect(wrapper.find('.clicks').text()).toBe('-1');

        wrapper.find('#b2').simulate('click');
        wrapper.find('#b2').simulate('click');

        expect(wrapper.find('.clicks').text()).toBe('1');
    });
});