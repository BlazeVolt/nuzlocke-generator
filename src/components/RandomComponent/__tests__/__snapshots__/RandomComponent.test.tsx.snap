// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RandomComponentBase /> renders the clicks section 1`] = `
<div
  className="clicks css-14ghush"
>
  0
</div>
`;
